Web.md

```diff
@@ -154,6 +154,78 @@ Transcoding failure is detected by `decodeURIComponent`, which may throw
 rule is only mandatory for Web embedding.
 
+## Source Maps for WebAssembly files on the web
+
+### Background
+[Source maps](https://docs.google.com/document/d/1U1RGAehQwRypUTovF1KRlpiOFze0b-_2gc6fAH0KY0k)
+are a debug-information format used on the web, generated by toolchains that target
+JavaScript, either from other languages (e.g. TypeScript, Emscripten) or from
+JavaScript itself (e.g. Closure, minifiers). A browser that supports source maps
+can display the original source in its developer tools instead of the compiled
+JavaScript that the VM actually executes. It can support setting breakpoints
+specified as source lines, single-stepping through source lines, and showing
+source lines in stack traces; however it cannot support getting or setting
+variables from the original source code because the format is not "full" debug
+info: it only maps locations from the compiled code back to locations in the
+source files (where a generated-code location consists of a line and column
+number, and a source location consists of file, line, and column). Generated
+files use a specially-formatted comment to specify a URL for a source map that
+describes the file.
+
+Source maps can be made to serve the same purpose for WebAssembly binary files,
+allowing parity with asm.js. By re-using an existing standard, browsers can
+take advantage of their existing source map support without having to make
+significant changes to their developer tools.
+
+
+### Alternatives
+
+An alternative method is to specify location mapping information in some way
+other than source maps. For example, it could be encoded directly into the wasm
+binary the way function and local names are written in the "name" section. This
+is more convenient if the VM itself uses the information (for example to
+generate a stack trace). However existing developer tools implementations are
+designed around the source map model of passing a URL from the VM to the dev
+tools code, which fetches and interprets the information. Therefore it is
+epxected that it will be much simpler to use "real" source maps than to design
+another format.
+
+Source maps also have known limitations (e.g. source variables cannot be
+described). A complete debug info format will eventually be needed; however the
+intention of this specification is to allow for easy integration with existing
+developer tools implementations in browsers.
+
+
+### Spec
+
+With few exceptions, most aspects of the source map
+[spec](https://docs.google.com/document/d/1U1RGAehQwRypUTovF1KRlpiOFze0b-_2gc6fAH0KY0k/)
+may be applied directly to wasm.
+Because wasm is a binary format (and thus does not have lines and columns, or
+comments per se), some simple reinterpretation of concepts is needed.
+
+#### Locations
+
+Source maps specify how to map a (zero-based) line and column position in
+generated code to a file, line, and column location in the source. For
+wasm binary locations, the line number is always 0, and the column number
```

Comment:
You could use the generated "line number" as the function index instead (or perhaps just the defined function index, skipping imported functions), and the generated "column" as the instruction offset in the function. But I suppose it's best to align w/ what's being discussed [here](https://github.com/WebAssembly/design/issues/990).

Is it worth mentioning this in some rationale?